lib/SILOptimizer/Mandatory/Differentiation.cpp

```diff
@@ -2164,33 +2164,29 @@ static SILFunction *getOrCreateReabstractionThunk(SILOptFunctionBuilder &fb,
     return alloc;
   };
 
-  // Store reabstracted results.
-  SmallVector<std::pair<unsigned, SILValue>, 4> directToIndirectResults;
-  SmallVector<SILValue, 4> indirectToDirectResults;
-
-  // Reabstract results.
-  assert(fromType->getResults().size() == toType->getResults().size());
-  for (unsigned resIdx : range(toType->getResults().size())) {
+  // Handle indirect results.
+  assert(fromType->getNumResults() == toType->getNumResults());
+  for (unsigned resIdx : range(toType->getNumResults())) {
     auto fromRes = fromConv.getResults()[resIdx];
     auto toRes = toConv.getResults()[resIdx];
     // No abstraction mismatch.
-    if (fromRes.isFormalIndirect() == toRes.isFormalIndirect())
+    if (fromRes.isFormalIndirect() == toRes.isFormalIndirect()) {
+      // If result typess are indirect, directly pass as next argument.
```

Comment:
```suggestion
      // If result types are indirect, directly pass as next argument.
```

Reply:
Thanks for the catch. Fixed in https://github.com/apple/swift/pull/22944/commits/c3beaef04711829186344f6c27bf43072e673412.